docs/machine-learning/how-to-guides/prepare-data-ml-net.md

```diff
@@ -0,0 +1,290 @@
+
+// Build ITransformer that performs operation
+var binningTransformer = binningEstimator.Fit(data);
+
+// Transform Data
+IDataView transformedData = binningTransformer.Transform(data);
+```
+
+The result of binning creates bin bounds of `[0,200000,Infinity]`. Therefore the resulting bins are `[0,1,1]` because the first observation is between 0-200000 and the others are greater than 200000 but less than infinity.
+
+## Work with categorical data
+
+Non-numeric categorical data needs to be converted to a number before being used to build a machine learning model. 
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+CarData[] cars = new CarData[] 
+{
+    new CarData
+    {
+        Color="Red",
+        VehicleType="SUV"
+    },
+    new CarData
+    {
+        Color="Blue",
+        VehicleType="Sedan"
+    },
+    new CarData
+    {
+        Color="Black",
+        VehicleType="SUV"
+    }
+};
+```
+
+The categorical `VehicleType` property can be converted into a number using the [`OneHotEncoding`](xref:Microsoft.ML.CategoricalCatalog.OneHotEncoding*) method. 
+
+```csharp
+// Define categorical transform estimator
+var categoricalEstimator = mlContext.Transforms.Categorical.OneHotEncoding("VehicleType");
+
+// Build ITransformer that performs operation
+ITransformer categoricalTransformer = categoricalEstimator.Fit(data);
+
+// Transform Data
+IDataView transformedData = categoricalTransformer.Transform(data);
+```
+
+The resulting transform converts the text value of `VehicleType` to a number. The entries in the `VehicleType` column become the following when the transform is applied: 
+
+```text
+[
+    1, // SUV
+    2, // Sedan
+    1 // SUV
+]
+```
+
+## Work with text data
+
+Text data needs to be transformed into numbers before using it to build a machine learning model. Visit the [transforms page](../resources/transforms.md) for a more detailed list and description of text transforms.
+
+Using the following input data which is loaded into an [`IDataView`](xref:Microsoft.ML.IDataView):
+
+```csharp
+ReviewData[] reviews = new ReviewData[]
+{
+    new ReviewData
+    {
+        Description="This is a good product",
+        Rating=4.7f
+    },
+    new ReviewData
+    {
+        Description="This is a bad product",
+        Rating=2.3f
+    }
+};
+```
+
+The minimum step to convert text to a numerical vector representation is to use the [`FeaturizeText`](xref:Microsoft.ML.TextCatalog.FeaturizeText*) method.
+
+```csharp
+// Define text transform estimator
+var textEstimator  = mlContext.Transforms.Text.FeaturizeText("Description");
+
+// Build ITransformer that performs operation
+ITransformer textTransformer = textEstimator.Fit(data);
+
+// Transform data
+IDataView transformedData = textTransformer.Transform(data);
+```
+
+The resulting transform would convert the text values in the `Description` column to a numerical vector that looks similar to the output below:
```

Comment:
>numerical vector [](start = 87, length = 16)

if you wanted to add a bit more info, you could put:

a numerical vector representing the normalized counts of the n-grams (groups of characters) appearing in the text.